test/framework/machinepool_helpers.go

```diff
@@ -286,16 +287,20 @@ func getMachinePoolInstanceVersions(ctx context.Context, input GetMachinesPoolIn
 	versions := make([]string, len(instances))
 	for i, instance := range instances {
 		node := &corev1.Node{}
-		err := wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
-			err := input.WorkloadClusterGetter.Get(ctx, client.ObjectKey{Name: instance.Name}, node)
-			if err != nil {
+		var nodeGetError error
+		err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			nodeGetError = input.WorkloadClusterGetter.Get(ctx, client.ObjectKey{Name: instance.Name}, node)
+			if nodeGetError != nil {
 				return false, nil //nolint:nilerr
 			}
 			return true, nil
 		})
 		if err != nil {
-			// Dump the instance name and error here so that we can log it as part of the version array later on.
-			versions[i] = fmt.Sprintf("%s error: %s", instance.Name, err)
+			versions[i] = "unknown"
+			if nodeGetError != nil {
```

Comment:
Just to double check. If nodeGetError is nil we don't want to surface err, right? I assume because it's just a timeout anyway?

Reply:
Yeah the error here is just a timeout, and previous to https://github.com/kubernetes-sigs/cluster-api/pull/8728 we didn't pass it along.

Reply:
I've just wrapped the error as it might be useful for tracing at some point.